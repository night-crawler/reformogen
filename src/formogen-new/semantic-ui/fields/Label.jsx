import React from 'react';
import PropTypes from 'prop-types';

import _ from 'lodash';

import { Popup } from 'semantic-ui-react';


SUILabel.propTypes = {
    upperFirstLabel: PropTypes.bool,
    verbose_name: PropTypes.string.isRequired,
    help_text: PropTypes.string.isRequired,
    // show help text on hover label (?) sign, or put in the bottom
    helpTextOnHover: PropTypes.bool,
};
export default function SUILabel(props) {
    const labelText = props.upperFirstLabel ? _.upperFirst(props.verbose_name) : props.verbose_name;
    if (!props.help_text || !props.helpTextOnHover) {
        return <label>{ labelText }</label>;
    }
    /* replaced trigger's <Icon name="help" /> with <i> to prevent react warning:
     * Warning: Stateless function components cannot be given refs. Attempts to access this ref will fail.
     */
    return (
        <label>
            <Popup
                trigger={ <i className='icon help' /> }

                content={ props.help_text }
                wide={ true }
                position='bottom left'
            />
            { labelText }
        </label>
    );
}
