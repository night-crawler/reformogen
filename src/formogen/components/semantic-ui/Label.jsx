import React from 'react';
import { Popup, Icon } from 'semantic-ui-react';
import _ from 'lodash';
import propTypes from '../fieldPropTypes';


SUILabel.propTypes = propTypes;
export default function SUILabel(props) {
    const labelText = props.upperFirstLabel ? _.upperFirst(props.verbose_name) : props.verbose_name;
    if (!props.help_text || !props.helpTextOnHover) {
        return <label>{ labelText }</label>;
    }
    return (
        <label>
            <Popup
                trigger={ <Icon name="help" /> }
                content={ props.help_text }
                wide={ true }
                position='bottom left'
            />
            { labelText }
        </label>
    );
}
