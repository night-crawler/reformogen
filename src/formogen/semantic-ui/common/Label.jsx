import React from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash';
import { Popup } from 'semantic-ui-react';


LabelComponent.displayName = 'Label';
LabelComponent.propTypes = {
  upperFirstLabel: PropTypes.bool,
  verbose_name: PropTypes.string.isRequired,
  help_text: PropTypes.string,
  /** show help text on hover label (?) sign, or put in the bottom */
  helpTextOnHover: PropTypes.bool,
};
function LabelComponent(props) {
  const labelText = props.upperFirstLabel ? _.upperFirst(props.verbose_name) : props.verbose_name;
  if (!props.help_text || !props.helpTextOnHover) {
    return <label>{ labelText }</label>;
  }
  /* replaced trigger's <Icon name="help" /> with <i> to prevent react warning:
     * Warning: Stateless function components cannot be given refs. Attempts to access this ref will fail.
  */
  return (
    <label>
      <Popup
        trigger={ <i className='icon help' /> }
        content={ props.help_text }
        wide={ true }
        position='bottom left'
      />
      { labelText }
    </label>
  );
}

export default LabelComponent;
