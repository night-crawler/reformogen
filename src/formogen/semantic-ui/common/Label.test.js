import Label from 'formogen/semantic-ui/common/Label';


it('should render Label', () => {
    const wrapper = shallow(
        <Label
            type='Label'
            name='Label: name'
            verbose_name='Label: verbose_name'
            help_text='Label: help_text'
            layoutOpts={ { width: 1 } }
        />
    );
    expect(wrapper).toMatchSnapshot();
});
