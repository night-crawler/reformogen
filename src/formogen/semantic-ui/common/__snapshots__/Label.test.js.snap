// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Label 1`] = `
<label>
  Label: verbose_name
</label>
`;
